src/Level Bars/main.vala

```diff
@@ -0,0 +1,64 @@
+#! /usr/bin/env -S vala workbench.vala --pkg gtk4
+
+Gtk.Entry entry;
+Gtk.LevelBar bar_discrete;
+Gtk.Label label_strength;
+
+public void main () {
+
+    var bar_continuous = (Gtk.LevelBar) workbench.builder.get_object ("bar_continuous");
+
+    bar_continuous.add_offset_value ("full", 100);
+    bar_continuous.add_offset_value ("half", 50);
+    bar_continuous.add_offset_value ("low", 25);
+
+    bar_discrete = (Gtk.LevelBar) workbench.builder.get_object ("bar_discrete");
+
+    bar_discrete.add_offset_value ("very-weak", 1);
+    bar_discrete.add_offset_value ("weak", 2);
+    bar_discrete.add_offset_value ("moderate", 4);
+    bar_discrete.add_offset_value ("strong", 6);
+
+
+    entry = (Gtk.Entry) workbench.builder.get_object ("entry");
+
+    label_strength = (Gtk.Label) workbench.builder.get_object ("label_strength");
+
+    entry.notify["text"].connect (estimate_password_strength);
+}
+
+// This is not a secure way to estimate password strength
+// Use appropriate solutions instead
+// such as https://github.com/dropbox/zxcvbn
+
+public void estimate_password_strength () {
+    var level = Math.fmin(Math.ceil(entry.text.length / 2), 6);
```

Comment:
Perform the cast to `int` here, so the type of `level` is clear, as mentioned in the style guide.
```suggestion
    var level = (int) Math.fmin (Math.ceil (entry.text.length / 2), 6);
```

Reply:
sorry about that missed that detail